File: frontend/src/app/model/FormInput.ts
import {FormLayout} from  './FormLayout';
import {FormInputType} from "./FormInputType";
import {FormValue} from "./FormValue";
import { MultipleValue } from './MultipleValue';
// export interface FormInput {
//   id: number;
//   title: string;
//   type: FormInputType;
//   formLayout?: FormLayout;
//   option: string;
//   required:boolean;
//   formValue?: FormValue;
  
// }
export interface FormInput {
  id: number;
  title: string;
  type: string;
  required: boolean;
  ordinalPosition?: number;
  config?: string;
  option?: string;
  formLayout: {
    id: number;
    title: string;
    type: string;
    parent?: FormLayout;
    children: any[];
  };
  formValue: any;
  multipleValues?: MultipleValue[];
}

File: backend/form-service/src/main/java/com/example/formservice/entities/FormInput.java
package com.example.formservice.entities;
import com.example.formservice.entities.enums.FormInputType;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.util.List;

@Data
@Entity
@EqualsAndHashCode(exclude = {"formLayout", "multipleValues"})
@ToString(exclude = {"formLayout", "multipleValues"})
public class FormInput {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Enumerated(EnumType.STRING)
    private FormInputType type;

    private boolean required;
    
    @Column(name = "ordinal_position")
    private Integer ordinalPosition = 0;

    @ManyToOne
    @JoinColumn(name = "form_layout_id")
    private FormLayout formLayout;

    @OneToMany(mappedBy = "formInput", cascade = CascadeType.ALL)
    @JsonManagedReference
    private List<MultipleValue> multipleValues;
}
